apps/nova/src/app/[locale]/(dashboard)/profile/teams/[teamId]/page.tsx

```
@@ -120,26 +171,87 @@
+    // Process submissions to calculate user scores correctly
+    const userProblemBestScores = new Map<string, Map<string, number>>();
+    const userChallengeScores = new Map<string, Map<string, number>>();
+
+    // First process all submissions to get best scores per problem per user
+    submissionsData.forEach((submission) => {
+      const userId = submission.user_id;
+      const problemId = submission.problem_id;
+      if (!userId || !problemId) return;
+
+      // Calculate the score properly according to the formula
+      const hasCriteria = submission.total_criteria ?? 0 > 0;
```

Biome (1.9.4)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)